### ISSUE TITLE
The run xxx must be in an active lifecycle_stage

### ISSUE THREAD

**Issue by agilebean** [mlflow/mlflow#689]
### System information
- **OS Platform and Distribution **: OSX 14.2 (Latest release)
- **MLflow installed from (source or binary)**: source
- **MLflow version (run ``mlflow --version``)**: 0.7.0
- **Python version**:  3.6
- **npm version (if running the dev UI):
- **Exact command to reproduce**:

```
mlflow.log_param("no_layers", 1)
mlflow.log_param("no_hidden_units", 16)
mlflow.log_param("activation", "relu")
mlflow.log_param("optimizer", "adam")
mlflow.log_metric("loss -testing", results[0])
mlflow.log_metric("acc -testing", results[1])
```

### Describe the problem
Train a model - succesfully. Log model parameters and metrics with above instructions - succesfully.
Then delete run in mlflow ui.
After deleting this run, repeating the above steps throws the exception:
```
MlflowException: The run 3df3bdc16d334463b5298fe697ba905d must be in an active lifecycle_stage.
```
### Source code / logs
model = models.Sequential()
model.add(layers.Dense(16, activation='relu', input_shape=(10000,)))
model.add(layers.Dense(1, activation='sigmoid'))

model.compile(
              optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'])

history = model.fit(x_train, y_train, epochs=4, batch_size=512)
results = model.evaluate(x_test, y_test)

**Comment by agilebean** [comment]
i figured out the error is not thrown if you precede the log_param() and log_metric() calls by 
```
with mlflow.start_run():
```
However, would be great for new users if the documentation was more explicit about how this exactly works.

**Comment by agilebean** [comment]
New: The same error message occurs after deleting the run in the mlflow ui, and calling 
`with mlflow.start_run():` again.
`mlflow.end_run()` doesn't kill the currently active run. 

How can you do that??

**Comment by mparkhe** [comment]
Hi @agilebean : Exception when logging metrics, params, ... etc to deleted runs is expected behavior if the run has been deleted. Deleted runs are moved to `.trash` folder under the experiment.

For the other question:
Yes, `mlflow.end_run()` will automatically end it. However it does not "delete it", the system will mark the status of the runs as ended with (success, failure... etc). When using `with` syntax `end_run` is called automatically for the started run.

However, the issue seems to be that the log statements seem to be attempting to log data for a deleted run. Please verify the following:
- This is not called within another run. If a run is active it is logged in `_active_run_stack`
- environment variable `MLFLOW_RUN_ID` is not set. in `mlflow.start_run` if this variable is set, system will look up that run uuid and set as active.

Ref: [Help for start_run](https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/fluent.py#L76)

**Comment by tomlincr** [comment]
As this issue is highly ranked when searching for "`mlflowexception: the run must be in 'active' lifecycle_stage.`", I've added another solution.

If the run has been deleted from the ui, but doesn't appear in `.trash`, you can manually edit the `meta.yaml` within the run folder (`mlruns/< UUID>/meta.yaml`) and change the value of the key `lifecycle_stage:` from `deleted` to `active`, this will then allow you to use `mlflow.end_run()`
